src/layout/Navbar/Navbar.svelte

```diff
@@ -167,20 +178,26 @@
 			{/if}
 		{/each}
 		<hr />
-		{#each buttons as { icon, href, label }}
-			<ListItem
-				{href}
+		{#each buttons as { name, path, external, icon, type }}
+			<a
+				class="item"
 				data-sveltekit-preload-data="hover"
-				type="navigation"
-				{...externalLink}
+				class:accent={type == "accent"}
+				class:selected={$page.url.pathname === path ||
+					($page.url.pathname.split("/").length > 1 &&
+						path.split("/").length > 1 &&
+						$page.url.pathname.startsWith(path) &&
+						!(path === "" || path === "/")) ||
+					(path === "/" && $page.url.pathname === "")}
+				href={path}
+				target={external ? "_blank" : undefined}
+				rel={external ? "noreferrer noopener" : undefined}
```

Comment:
Please convert this into a util function and document what it does. This complexity inline seems a bit too much.